sklearn/tree/tests/test_tree.py

```diff
@@ -1698,12 +1698,13 @@ def test_mae():
     dt_mae = DecisionTreeRegressor(random_state=0, criterion="mae",
                                    max_leaf_nodes=2)
     dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3])
-    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0/3.0])
+    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0 / 3.0])
     assert_array_equal(dt_mae.tree_.value.flat, [4, 4.5, 4.0])
 
     dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3],
                [0.6, 0.3, 0.1, 1.0, 0.3])
-    assert_array_equal(dt_mae.tree_.impurity, [7.0/2.3, 3.0/0.7, 4.0/1.6])
+    assert_array_almost_equal(dt_mae.tree_.impurity,
```

Comment:
we should probably add the case where we have the same sample_weight (array of one) which should give the previous results as well.

Reply:
Good point, added (with additional comments too)